packages/jest-matchers/src/__tests__/matchers-test.js

```diff
@@ -55,6 +55,21 @@ describe('.toBe()', () => {
     expect(() => jestExpect(obj).toBe({}))
       .toThrowErrorMatchingSnapshot();
   });
+
+  test(`assertion error matcherResult property contains 
+  matcher name, expected and actual values`, () => {
+    const actual = {a: 1};
+    const expected = {a: 2};
+    try {
+      jestExpect(actual).toBe(expected);
+    } catch (error) {
+      const matcherResult = error.matcherResult;
+      expect(matcherResult).toBeTruthy();
```

Comment:
I would use 1 asymmetric matcher `expect.objectContaining` instead of 4 normal ones, but I think it's not a big deal.